Sorting/TimSort.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * Tim Sort
+ * 
+ * @param array $input array of random numbers
+ * @return array sorted array of numbers
+ */
+function timSort(array $input)
+{
+    $length = count($input);
+    $runs = [];
+    $sorted_runs = [];
+    $new_run = [$input[0]];
+    $sorted_array = [];
+    $i = 1;
+
+    while ($i < $length) {
+        if ($input[$i] < $input[$i - 1]) {
+            $runs[] = $new_run;
+            $new_run = [$input[$i]];
+        } else {
+            $new_run[] = $input[$i];
+        }
+
+        $i++;
+    }
+
+    $runs[] = $new_run;
+    
+
+    foreach ($runs as $run) {
+        $sorted_runs[] = insertionSort($run);
+    }
+
+    foreach ($sorted_runs as $run) {
+        $sorted_array = merge($sorted_array, $run);
+    }
+
+    return $sorted_array;
+}
+
+/**
+ * Bynary Search
+ * 
+ * @param array $input
+ * @param int $value
+ * @return int
+ */
+function binarySearch(array $input, int $value, int $left, int $right)
+{
+    if ($right < $left) {
+        return -1;
+    }
+
+    $mid = floor(($left + $right) / 2);
+
+    if ($input[$mid] > $value) {
+        return binarySearch($input, $value, $left, $mid - 1);
+    } elseif ($input[$mid] < $value) {
+        return binarySearch($input, $value, $mid + 1, $right);
+    } else {
+        return $mid;
+    }
+}
```

Comment:
This function already exists.  It's defined in Searches.  No need to duplicate the function.